src/diff-snapshot.js

```diff
@@ -13,63 +13,60 @@
  */
 
 const fs = require('fs');
-const BlinkDiff = require('blink-diff');
-const intersection = require('lodash/intersection');
+const PNG = require('pngjs').PNG;
+const pixelmatch = require('pixelmatch');
 const mkdirp = require('mkdirp');
 const path = require('path');
 
-const unsupportedDiffConfigKeys = [
-  'imageAPath',
-  'imageA',
-  'imageBPath',
-  'imageB',
-  'imageOutputPath',
-];
-
-function isDiffConfigValid(customDiffConfig) {
-  return intersection(unsupportedDiffConfigKeys, Object.keys(customDiffConfig)).length === 0;
-}
-
 function diffImageToSnapshot(options) {
   const {
     imageData,
     snapshotIdentifier,
     snapshotsDir,
     updateSnapshot = false,
     customDiffConfig = {},
-   } = options;
-
-  if (!isDiffConfigValid(customDiffConfig)) {
-    throw new Error(
-      `Passing in options: ${unsupportedDiffConfigKeys} via Blink-Diff configuration `
-      + 'is not supported as those option are internally used. '
-      + 'Instead pass your image data as first argument to this function!'
-    );
-  }
+  } = options;
 
   let result = {};
   const baselineSnapshotPath = path.join(snapshotsDir, `${snapshotIdentifier}-snap.png`);
   if (fs.existsSync(baselineSnapshotPath) && !updateSnapshot) {
     const outputDir = path.join(snapshotsDir, '__diff_output__');
     const diffOutputPath = path.join(outputDir, `${snapshotIdentifier}-diff.png`);
-    const defaultBlinkDiffConfig = {
-      imageA: imageData,
-      imageBPath: baselineSnapshotPath,
-      thresholdType: 'percent',
+    const defaultDiffConfig = {
       threshold: 0.01,
-      imageOutputPath: diffOutputPath,
     };
 
     mkdirp.sync(outputDir);
-    const diffConfig = Object.assign({}, defaultBlinkDiffConfig, customDiffConfig);
-    const diff = new BlinkDiff(diffConfig);
-    const unformattedDiffResult = diff.runSync();
+    const diffConfig = Object.assign({}, defaultDiffConfig, customDiffConfig);
 
-    result = Object.assign(
-      {},
-      unformattedDiffResult,
-      { diffOutputPath }
+    const comparisonImg = PNG.sync.read(imageData);
+    const baselineImg = PNG.sync.read(fs.readFileSync(baselineSnapshotPath));
+
+    const diffImg = new PNG({ width: comparisonImg.width, height: comparisonImg.height });
+    const pixelCountDiff = pixelmatch(
+      comparisonImg.data,
+      baselineImg.data,
+      diffImg.data,
+      comparisonImg.width,
+      comparisonImg.height,
+      diffConfig
     );
+
+    const totalPixels = comparisonImg.width * comparisonImg.height;
+    const percentDiff = pixelCountDiff / totalPixels;
```

Comment:
Like I said before shouldnt `percentDiff` be renamed to `diffRatio` or something like that? Since it is not actually a percentage...

Reply:
I went with your previous suggestion. https://github.com/americanexpress/jest-image-snapshot/pull/17#discussion_r145316914

Reply:
Right but that's an outdated comment from another PR. What I am wanting to resolve here is the fact that `percentDiff` implies that the value is a percentage (i.e. 90%). However what that variable actually stores is more of a ratio (i.e. `0.9`). and so renaming the variable would help with readability.

Reply:
Done